trainer/derrian_backend/sd_scripts/sdxl_train.py

```
@@ -618,11 +870,159 @@ def optimizer_hook(parameter: torch.Tensor):
+                    latents = latents - args.vae_shift_factor
+                latents = latents * args.vae_scale_factor
+
+                text_encoder_conds = []
+                masks_reshaped = []
+                text_encoder_outputs_list = batch.get("text_encoder_outputs_list", None)
+                if text_encoder_outputs_list is not None:
+                    text_encoder_conds = text_encoder_outputs_list
+                    masks_reshaped = text_encoder_outputs_list[3:]
+
+                if len(text_encoder_conds) == 0 or text_encoder_conds[0] is None:
+                    with accelerator.autocast():
```

Ruff (0.15.12)

[error] Undefined name `accelerator`

(F821)